### ISSUE TITLE
[BUG] 401 "claim is empty or token is invalid" when using Azure.Identity

### ISSUE THREAD

**Issue by MarienMonnier** [Azure/azure-sdk-for-net#20485]
**Describe the bug**
When I'm using Azure.Identity authentication, I can't get all topics & all queues from a service bus instance. The first call to get the queues will succeed but the second call to get the topics will always fail (or the other way around).

**Expected behavior**
Can get all topics & all queues from a service bus instance.

**Actual behavior (include Exception or Stack Trace)**

The second call to get all queues/topics will fail:
```
System.UnauthorizedAccessException
  HResult=0x80070005
  Message=claim is empty or token is invalid. TrackingId:7f293702-89fa-46ca-a04e-012f6104b863_G25, SystemTracker:xxx.servicebus.windows.net:$Resources/queues, Timestamp:2021-04-19T07:25:04
Status: 401 (Unauthorized)

Content:
<Error><Code>401</Code><Detail>claim is empty or token is invalid. TrackingId:7f293702-89fa-46ca-a04e-012f6104b863_G25, SystemTracker:xxx.servicebus.windows.net:$Resources/queues, Timestamp:2021-04-19T07:25:04</Detail></Error>

Headers:
Transfer-Encoding: chunked
Server: Microsoft-HTTPAPI/2.0
Strict-Transport-Security: REDACTED
Date: Mon, 19 Apr 2021 07:25:04 GMT
Content-Type: application/xml; charset=utf-8

  Source=Azure.Messaging.ServiceBus
  StackTrace:
   at Azure.Messaging.ServiceBus.Administration.HttpRequestAndResponse.<ThrowIfRequestFailedAsync>d__7.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Azure.Messaging.ServiceBus.Administration.HttpRequestAndResponse.<SendHttpRequestAsync>d__14.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Azure.Messaging.ServiceBus.Administration.HttpRequestAndResponse.<GetEntityAsync>d__11.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Azure.Messaging.ServiceBus.Administration.HttpRequestAndResponse.<GetEntitiesPageAsync>d__10`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at Azure.Core.PageResponseEnumerator.FuncAsyncPageable`1.<AsPages>d__2.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult(Int16 token)
   at Azure.Core.PageResponseEnumerator.FuncAsyncPageable`1.<AsPages>d__2.System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult(Int16 token)
   at Azure.AsyncPageable`1.<GetAsyncEnumerator>d__6.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Azure.AsyncPageable`1.<GetAsyncEnumerator>d__6.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult(Int16 token)
   at Azure.AsyncPageable`1.<GetAsyncEnumerator>d__6.System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult(Int16 token)
   at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()
   at Xxx.TestController.<Test>d__0.MoveNext() in C:\Dev\repos\xxx\TestController.cs:line 51

Inner Exception 1:
RequestFailedException: claim is empty or token is invalid. TrackingId:7f293702-89fa-46ca-a04e-012f6104b863_G25, SystemTracker:xxx.servicebus.windows.net:$Resources/queues, Timestamp:2021-04-19T07:25:04
Status: 401 (Unauthorized)

Content:
<Error><Code>401</Code><Detail>claim is empty or token is invalid. TrackingId:7f293702-89fa-46ca-a04e-012f6104b863_G25, SystemTracker:xxx.servicebus.windows.net:$Resources/queues, Timestamp:2021-04-19T07:25:04</Detail></Error>

Headers:
Transfer-Encoding: chunked
Server: Microsoft-HTTPAPI/2.0
Strict-Transport-Security: REDACTED
Date: Mon, 19 Apr 2021 07:25:04 GMT
Content-Type: application/xml; charset=utf-8
```

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Add your user to the "Azure Service Bus Data Owner" role

2. Try to get all topics & all queues of this service bus instance and the second call will fail.

For example, if I first get all topics than all queues, the call to get all queues will fail with a 401.
Here is a sample:
```csharp
var administrationClient = new ServiceBusAdministrationClient("xxx.servicebus.windows.net", new VisualStudioCredential());
var topics = administrationClient.GetTopicsAsync();
await foreach (var topic1 in topics)
{
}

var queues = administrationClient.GetQueuesAsync(); // this line will fail with a 401
await foreach (var queue1 in queues)
{
}
```

If I try the other way around, the call to `GetQueuesAsync` will succeed but the call to `GetTopicsAsync` will fail with a 401.
I also tried to create a new instance of the `ServiceBusAdministrationClient` & of the `TokenCredential` (either `DefaultCredential` or `VisualStudioCredential`), but it still fails.

Please note that I tried with the old `ManagementClient`, and I have the same issue:
```csharp
var managementClient = new ManagementClient("sb://xxx.servicebus.windows.net", new ManagedIdentityTokenProvider());
var topics = await managementClient.GetTopicsAsync();
var queues = await managementClient.GetQueuesAsync(); // 401 Unauthorized
```

Exception:
```
Microsoft.Azure.ServiceBus.UnauthorizedException
  HResult=0x80131500
  Message=claim is empty or token is invalid. TrackingId:7b487659-cd9d-4ad1-ae79-117899444f5b_G18, SystemTracker:xxx.servicebus.windows.net:$Resources/queues, Timestamp:2021-04-19T07:33:32
  Source=Microsoft.Azure.ServiceBus
  StackTrace:
   at Microsoft.Azure.ServiceBus.Management.ManagementClient.<SendHttpRequest>d__53.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.ServiceBus.Management.ManagementClient.<GetEntity>d__50.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.ServiceBus.Management.ManagementClient.<GetQueuesAsync>d__21.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Xxx.TestController.<Test>d__0.MoveNext() in C:\Dev\repos\Xxx\TestController.cs:line 37

  This exception was originally thrown at this call stack:
    [External Code]
    Xxx.TestController.Test() in TestController.cs
```

I also tried to create a new service bus instance with a few queues & topics, but it still fails.

**Environment:**
 - Name and version of the Library package used: Azure.Messaging.ServiceBus 7.1.2 (and Microsoft.Azure.ServiceBus 5.1.2 for the ManagementClient)
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): Windows 10 - .NET Core 3.1
```
.NET SDK (reflecting any global.json):
 Version:   5.0.200
 Commit:    70b3e65d53

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.200\

Host (useful for support):
  Version: 5.0.3
  Commit:  c636bbdc8a

.NET SDKs installed:
  2.1.801 [C:\Program Files\dotnet\sdk]
  2.1.802 [C:\Program Files\dotnet\sdk]
  2.2.401 [C:\Program Files\dotnet\sdk]
  2.2.402 [C:\Program Files\dotnet\sdk]
  5.0.101 [C:\Program Files\dotnet\sdk]
  5.0.200 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```
 - IDE and version : Visual Studio 16.9.0

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by MarienMonnier** [comment]
Hello, any news about this issue? This is a blocker for us to use User Managed Identity instead of access keys.

**Comment by ramya-rao-a** [comment]
@MarienMonnier Apologies for the delay in response here
@JoshLove-msft Can you please review this issue? Managed Identity should work as expected with the Service Bus library as far as I know. So, may be some set up problems?

**Comment by JoshLove-msft** [comment]
Yes, I reproed the issue and I believe @yvgopal is going to look into it.

**Comment by MrKevHunter** [comment]
This is an issue for us too, seems like a fairly common use case

**Comment by sreem007** [comment]
Hi Team, 
Can we please have an update on this bug/issue.

**Comment by shankarsama** [comment]
We are actively investigating this issue. We will provide an update asap. cc @serkantkaraca

**Comment by sa1nt** [comment]
We ran into this using `com.azure:azure-messaging-servicebus:7.6.0` library in Java on a new Service Bus namespace in West Europe.

In our case we tried to use a _single_ `ServiceBusAdministrationAsyncClient` to create-if-missing _several_ ASB Topics, and started getting `HTTP 401 "claim is empty or token is invalid"` errors somewhere in the middle. 

Workaround for us was to create a new `ServiceBusAdministrationAsyncClient` for every ASB Topic we had to check and create if necessary.

**Comment by navba-MSFT** [comment]
@MarienMonnier Apologies for the late reply. Could you please confirm if you are still facing this issue ? Awaiting your reply.

**Comment by MarienMonnier** [comment]
Hi @navba-MSFT ,

I hadn't test since the last time. I can confirm that this is now working correctly like @MrKevHunter said.
I tested with the newest versions (Azure.Messaging.ServiceBus 7.7.0 & Microsoft.Azure.ServiceBus 5.2.0) and the versions I used a year ago when I faced the issue (Azure.Messaging.ServiceBus 7.1.2 & Microsoft.Azure.ServiceBus 5.1.2) and I confirm that this is working correctly, so the fix must have been made on Service Bus side and not inside the SDK.

Thanks,
